Classifier and Summarizer/WebEntityDetection.py

```diff
@@ -0,0 +1,120 @@
+    Return type:
+        list : of images encoded in the above format
+    '''
+    base64Encodings = []
+    
+    for imagePath in images:
+      with open(imagePath,'rb') as f:
+        encodedImage = base64.encodebytes(f.read()).decode()
+        base64Encodings.append(encodedImage)
+    
+    return base64Encodings
+
+  def formatSingleRequest(self,data:str) ->dict:
+    '''
+    formats the request as a dict 
+    with the required data and returns it
+    Params :
+      data : a base64-encoded image as a utf-8 string
+    Return type:
+      dict : formatted as shown below
+    '''
+    return {
+      "image":{
+        "content" : data
+      },
+      "features":[
+        {
+          "maxResults": str(self.maxEntities),
+          "type": "WEB_DETECTION"
+        }
+      ],
+      "imageContext": {
+          "webDetectionParams": {
+            "includeGeoResults": "true"
+          }
+      }
+    }
+
+  def prepareRequests(self,encodedImages:list) ->list:
+    '''
+    prepares the requests for making calls
+    to the api. Structures the given images
+    in the format as expected by the API
+    Params :
+      encodedImages : numpy.ndarray of base64 encoded images
+    Return Type:
+    list: of required format
+    '''
+    imageRequests = []
+    
+    for image in encodedImages:
+      imageRequests.append(self.formatSingleRequest(image))
+    
+    return imageRequests
+
+
+  def getBestGuessLabel(self,jsonResponse,imageIndex):
+    ''' return the best guess label '''
+    return jsonResponse["responses"][imageIndex]["webDetection"]["bestGuessLabels"][0]["label"]
+
+  def getTopEntities(self,jsonResponse,numberOfEntities,imageIndex):
+    ''' return the top entity description'''
+    topEntites = ""
+    numberOfEntities = min(numberOfEntities, len(jsonResponse["responses"][imageIndex]["webDetection"]["webEntities"]))
+    for i in range(numberOfEntities):
+      if "description" in jsonResponse["responses"][imageIndex]["webDetection"]["webEntities"][i]:
+        topEntites+=jsonResponse["responses"][imageIndex]["webDetection"]["webEntities"][i]["description"]+"."
+    return topEntites
+
+
+  def getDescriptionForImages(self,images:list) -> list:
+    '''
+    given a list of images returns the best guess
+    labels for the web entities or description of
+    the images
+    Params:
+      images : list of image paths
+    Return type:
+    list<str> : list containing best guess descriptions of the image
+    '''
+    base64EncodedImages = self.findBase64EncodingForImages(images)
+
+    imageRequests = self.prepareRequests(base64EncodedImages)
+    
+    numberOfImages = len(images)
+    
+    jsonDataForPostRequest = json.dumps({
+      "requests": imageRequests
+    })
+    response = requests.post(self.apiUrl,data=jsonDataForPostRequest)
+    if response.status_code!=200:
+      return []
+    response = json.loads(response.content)
+    imageDescriptions = []
+    for i in range(numberOfImages):
```

Comment:
Since this is the only place where "numberOfImages" is used, we can use "len(images)" directly here

Reply:
Updated accordingly!